homeassistant/components/unifiprotect/services.py

```diff
@@ -42,93 +38,78 @@
 )
 
 
-def _async_all_ufp_instances(hass: HomeAssistant) -> list[ProtectApiClient]:
+def _async_all_ufp_instances(hass: HomeAssistant) -> list[ProtectData]:
     """All active UFP instances."""
     return [
-        data.api for data in hass.data[DOMAIN].values() if isinstance(data, ProtectData)
+        data for data in hass.data[DOMAIN].values() if isinstance(data, ProtectData)
     ]
 
 
 @callback
-def _async_get_macs_for_device(device_entry: dr.DeviceEntry) -> list[str]:
-    return [
-        _async_unifi_mac_from_hass(cval)
-        for ctype, cval in device_entry.connections
-        if ctype == dr.CONNECTION_NETWORK_MAC
-    ]
-
-
-@callback
-def _async_get_ufp_instances(
-    hass: HomeAssistant, device_id: str
-) -> tuple[dr.DeviceEntry, ProtectApiClient]:
+def _async_get_ufp_instance(hass: HomeAssistant, device_id: str) -> ProtectData:
     device_registry = dr.async_get(hass)
     if not (device_entry := device_registry.async_get(device_id)):
         raise HomeAssistantError(f"No device found for device id: {device_id}")
 
     if device_entry.via_device_id is not None:
-        return _async_get_ufp_instances(hass, device_entry.via_device_id)
+        return _async_get_ufp_instance(hass, device_entry.via_device_id)
 
-    macs = _async_get_macs_for_device(device_entry)
     ufp_instances = [
-        i for i in _async_all_ufp_instances(hass) if i.bootstrap.nvr.mac in macs
+        i
+        for i in _async_all_ufp_instances(hass)
+        if i.async_get_ufp_device_from_device(device_id) is not None
     ]
 
     if not ufp_instances:
         # should not be possible unless user manually enters a bad device ID
         raise HomeAssistantError(  # pragma: no cover
-            f"No UniFi Protect NVR found for device ID: {device_id}"
+            f"No UniFi Protect Config Entry found for device ID: {device_id}"
         )
```

Comment:
You can cover this with 

```python3
async def test_set_default_doorbell_text_orphened_device(
    hass: HomeAssistant, device: dr.DeviceEntry, mock_entry: MockEntityFixture
):
    """Test set_default_doorbell_text service with an orphened device entry."""
    dev_reg = dr.async_get(hass)
    broken_device = dev_reg.async_get_or_create(
        identifiers={DOMAIN,"AABBCCDDEEFF"},
        config_entry_id=mock_entry.entry.entry_id
    )
    nvr = mock_entry.api.bootstrap.nvr
    nvr.__fields__["set_default_doorbell_message"] = Mock()
    nvr.set_default_doorbell_message = AsyncMock()

    with pytest.raises(HomeAssistantError):
        await hass.services.async_call(
            DOMAIN,
            SERVICE_SET_DEFAULT_DOORBELL_TEXT,
            {ATTR_DEVICE_ID: broken_device.id, ATTR_MESSAGE: "Test Message"},
            blocking=True,
        )
```

Reply:
Confirmed it worked. That is great and it fixed two of the lines.